packages/core/content-manager/admin/src/pages/EditView/utils/data.ts

```diff
@@ -219,6 +219,25 @@ type HandleOptions = {
 
 type RemovedFieldPath = string;
 
+/**
+ * @internal
+ * @description Finds the initial value for a component or dynamic zone item (based on its __temp_key__ and not its index).
```

Comment:
I believe the `__temp_key__` is used for drag and drop and will change when you move an item. Will that have an impact?